### ISSUE TITLE
[fvt]2.12.1 xcatprobe image -c return errors if node could not using xdsh to get information

### ISSUE THREAD

**Issue by junxiawang** [xcat2/xcat-core#1428]
env：linux
build:
Version 2.12.1 (git commit fddd78e1ec6a9d59bc921bef24e300e8d94f69e4, built Sun Jun 19 06:45:32 EDT 2016)
How to reproduce:
```
c910f04x30v01:/opt/xcat/bin # ./xcatprobe  image -c
Smartmatch is experimental at ./xcatprobe line 64.
Smartmatch is experimental at ./xcatprobe line 173.
Use of uninitialized value $node_image_table in concatenation (.) or string at /opt/xcat/probe//subcmds/image line 216.
node is switch-10-4-25-1
[FAIL] Not all compute nodes are installed with the same os image.

```
```
1. This is a switch so no need to check the image
2. From code:
  my $node_image_table;
        foreach $compute_node (keys %node_running_image_name_hash) {
            $node_image_table .= sprintf("%-15s %-30s : %-20s\n", $compute_node, $node_running_image_name_hash{$compute_node}, $node_running_image_uuid_hash{$compute_node});
        }
        $msg = "Not all compute nodes are installed with the same os image.\n" . $node_image_table;
        $status = "f";
....
my $output = `xdsh $_ "cat /opt/xcat/xcatinfo"`;
------------------------------------>
 Image check could not get xcatinfo from the switch so it return errors.I think code should check whether it is diskless node and whether it could get information using xdsh
```

**Comment by gurevichmark** [comment]
Fixed by #1406

**Comment by junxiawang** [comment]
verified
